Dockerfile

```diff
@@ -0,0 +1,60 @@
+# Define function directory
+ARG FUNCTION_DIR="/function"
+
+# Base image of the docker container
+FROM node:lts-slim as build-image
+
+# Include global arg in this stage of the build
+ARG FUNCTION_DIR
+
+# AWS Lambda runtime dependencies
+RUN apt-get update && \
+    apt-get install -y \
+        g++ \
+        make \
+        unzip \
+        libcurl4-openssl-dev \
+        autoconf \
+        automake \
+        libtool \
+        cmake \
+        python3 \
+        libkrb5-dev
+
+# Copy function code
+RUN mkdir -p ${FUNCTION_DIR}/
+
+COPY opensearch-reporting-cli-1.0.0.tgz ${FUNCTION_DIR}/
```

Comment:
We can add download tar file here when it's available on downloads page.

Reply:
just an opinion, what's the benefit of downloading artifact in Dockerfile? I think most Dockerfile builds artifact from source, this way it's up to date. Users can check out tags if they want to use a specific version

Reply:
I think it minimizes no of steps to setup with lambda with latest cli version. But we can just add two sample Dockerfile. One with building from source code and another for getting latest released tar file and build image. What do you think?

Reply:
Sorry for the misunderstanding, I didn't mean the minimum steps as is.
We can get the file from opensearch artifacts instead of copy. I added copy since there is no artifact yet so it will fail.
```
RUN curl -LJO https://artifacts.opensearch.org/reporting-cli/opensearch-reporting-cli-1.0.0.tgz
```

So steps needed from user will be
1. Get Dockerfile from documentation page/ github
2. run docker build 

User don't need to have git installed in this case.

Reply:
i don't think git will be an issue since we require docker already, but they can always use wget/curl if they don't have git

```sh
wget https://github.com/opensearch-project/reporting-cli/archive/main.tar.gz
```

I think my main concerns are:

1. user needs to edit Dockerfile if they want to use another version
2. If a new bug fix got merged, the Dockerfile under the same commit will still build without the fix unless we release a new version, which is counterintuitive
3. we will need to bump version in Dockerfile, which is extra release step

They are minor issues so I'm ok with current setup, but if we go this way could you consider:

1. add docker `--build-arg` support for choosing version, it should default to a hard coded (latest) version if not specified
2. document how to build docker image from latest source code

Reply:
I totally agree. For now, I moved the Dockerfile to doc as it's purpose is to provide example. Also added doc to build file from source.